onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -454,6 +454,12 @@ SubGraphCollection_t TensorrtExecutionProvider::GetSupportedList(SubGraphCollect
         std::string string_buf;
         model_proto.SerializeToString(&string_buf);
 
+		if (dump_subgraphs_) {
+		  // Dump TensorRT subgraph for debugging if enabled via ORT_TENSORRT_DUMP_SUBGRAPHS env variable.
+		  std::fstream dump("TensorrtExecutionProvider_TRT_Subgraph.onnx", std::ios::out | std::ios::trunc | std::ios::binary);
```

Comment:
use meta_def->name instead? otherwise how do we know which one if they all had same file name.
btw, how would a trt parser error below show up?
does it throw an exception or return error?
it would be nice to be able to catch it, print an error message which can associate that error with the onnx file that was just dumped.  
(similar to the trt engine build error that we log in Compile() )

Reply:
indexedsubgraph is not created and no meta_def is available at this stage.
In some cases, trt_parser->supportsModel() failed to parse subgraphs and it doesn't return the error or exception to ORT. The idea here is to dump out only the subgraph that the parser failed on so that we can use it to debug into the problem.